celery-java/src/main/java/com/geneea/celery/Celery.java

```diff
@@ -166,6 +190,61 @@ private String getLocalHostName() {
         return new AsyncResultImpl<>(result);
     }
 
+    /**
+     * Submit a task by name with priority.
+     *
+     * @param name task name as understood by the worker
+     * @param priority the priority of the message
+     * @param args positional arguments for the method (need to be JSON serializable)
+     * @return asynchronous result
+     * @throws IOException
+     */
+    public AsyncResult<?> submitWithPri(String name, int priority, Object[] args) throws IOException {
```

Comment:
I'd prefer if this and other methods kept the same name as the existing ones, just overloaded by parameters.